File: lib/backend/src/services/NewsService.ts
export class NewsService {
  static async getHytaleNews() {
    try {
      const response = await fetch('https://launcher.hytale.com/launcher-feed/release/feed.json', {
        headers: {
          'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36'
        }
      });
      
      if (!response.ok) throw new Error('Failed to fetch news');

      const data = await response.json() as any;
      const articles = data.articles || [];
      
      return articles.map((article: any) => ({
        title: article.title || '',
        description: article.description || '',
        destUrl: article.dest_url || '',
        imageUrl: article.image_url ? 
          (article.image_url.startsWith('http') ? 
            article.image_url : 
            `https://launcher.hytale.com/launcher-feed/release/${article.image_url}`
          ) : '',
        date: article.published_at || '',
        tag: 'News'
      }));
    } catch (error: any) {
      console.error('Failed to fetch news:', error.message);
      return [];
    }
  }
}

File: lib/backend/src/services/NewsService.py
import requests
from .LoggerService import LoggerService

class NewsService:
    NEWS_URL = 'https://launcher.hytale.com/launcher-feed/release/feed.json'
    
    @staticmethod
    def get_hytale_news():
        try:
            LoggerService.info(f"Fetching news from {NewsService.NEWS_URL}")
            
            response = requests.get(
                NewsService.NEWS_URL,
                headers={
                    'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36',
                    'Accept': 'application/json'
                },
                timeout=10
            )
            
            if response.status_code != 200:
                LoggerService.warning(f"News API returned status {response.status_code}")
                return []
            
            data = response.json()
            articles = data.get('articles', [])
            LoggerService.info(f"Received {len(articles)} articles from Hytale")
            
            result = []
            for article in articles:
                image_url = article.get('image_url', '')
                if image_url and not image_url.startswith('http'):
                    image_url = f"https://launcher.hytale.com/launcher-feed/release/{image_url}"
                
                result.append({
                    "title": article.get('title', ''),
                    "description": article.get('description', ''),
                    "destUrl": article.get('dest_url', ''),
                    "imageUrl": image_url,
                    "date": article.get('published_at', ''),
                    "tag": "News"
                })
            
            LoggerService.info(f"Processed {len(result)} news articles")
            return result
            
        except requests.exceptions.Timeout:
            LoggerService.error("News API request timed out")
            return []
        except requests.exceptions.RequestException as e:
            LoggerService.error(f"News API request failed: {e}")
            return []
        except Exception as e:
            LoggerService.error(f"Failed to parse news: {e}")
            import traceback
            LoggerService.error(traceback.format_exc())
            return []
